ksp_plugin/flight_plan.hpp

```diff
@@ -0,0 +1,85 @@
+﻿#pragma once
+
+#include <vector>
+#include <stack>
+
+#include "ksp_plugin/burn.hpp"
+#include "ksp_plugin/fork_handle.hpp"
+#include "ksp_plugin/frames.hpp"
+#include "ksp_plugin/manœuvre.hpp"
+#include "ksp_plugin/plugin.hpp"
+#include "physics/discrete_trajectory.hpp"
+
+namespace principia {
+
+using integrators::AdaptiveStepSizeIntegrator;
+
+namespace ksp_plugin {
+
+using NavigationManœuvre = Manœuvre<Barycentric, Navigation>;
+
+class FlightPlan {
+ public:
+  FlightPlan(
+      not_null<DiscreteTrajectory<Barycentric>*> root,
+      Instant const& initial_time,
+      Instant const& final_time,
+      Mass const& initial_mass,
+      not_null<Ephemeris<Barycentric>*> ephemeris,
+      AdaptiveStepSizeIntegrator<
+          Ephemeris<Barycentric>::NewtonianMotionEquation> const& integrator,
+      Length const& length_integration_tolerance,
+      Speed const& speed_integration_tolerance);
+  ~FlightPlan() = default;
+
+  int size() const;
+  NavigationManœuvre const& Get(int index);
+
+  void RemoveLast();
+
+  // The following two functions return false and have no effect if the given
+  // |burn| would start before |initial_time_| or before the end of the previous
+  // burn, or end after |final_time_|.
+  bool Append(Burn burn);
+  bool ReplaceLast(Burn burn);
+
+  // Returns false and has no effect if |final_time| is before the end of the
+  // last manœuvre or before |initial_time_|.
+  bool SetFinalTime(Instant const& final_time);
+
+  void SetTolerances(
+      Length const& length_integration_tolerance,
+      Speed const& speed_integration_tolerance);
+
+ private:
+  void Append(NavigationManœuvre manœuvre);
```

Comment:
Comment what the private methods do.

Also, is there any reason for passing by copy and moving?  On the surface it would seem possible to copy a Manoeuvre.  Moving is fraught with difficulties with the object being left in a botched state, so I'd rather pass by const& and copy.